eox_nelp/course_experience/models.py

```diff
@@ -0,0 +1,114 @@
+"""
+Course experience models. This contains all the model related with the course user experience.
+
+Models:
+    LikeDislikeSection: Store user decision(like or dislike) for specific sub-sections(group of units).
+    LikeDislikeCourse: Store user decision(like or dislike) for a course.
+    ReportSection: Store report reason about a specific sub-section.
+    ReportCourse: Store report reason for a course.
+"""
+from django.contrib.auth import get_user_model
+from django.db import models
+from opaque_keys.edx.django.models import CourseKeyField, UsageKeyField
+
+User = get_user_model()
+
+
+class BaseLikeDislike(models.Model):
+    """Base abstract model for like and dislike records.
+
+    fields:
+        author<Foreignkey>: Makes reference to the user record associated with the status.
+        status<BooleanField>: True = Liked, False =d isliked and None = not-set
+    """
+    author = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
+    status = models.BooleanField(null=True)
+
+    class Meta:
+        """Set model abstract"""
+        abstract = True
+
+
+class BaseReport(models.Model):
+    """Base abstract model for reporting records.
+
+    fields:
+        author<Foreignkey>: Makes reference to the user record associated with the reason.
+        reason<CharField>: Store report reason as a code, e.g SC => Sexual Content.
+    """
+    REPORT_REASONS = [
+        ("SC", "Sexual content"),
+        ("GV", "Graphic violence"),
+        ("HA", "Hateful or abusive content"),
+        ("CI", "Copycat or impersonation"),
+        ("OO", "Other objection"),
```

Comment:
I thought another objection would be a variable or something the user could change or give more information about.
by the way, at the moment we can keep it this way.